.github/workflows/funcbench.yml

```diff
@@ -0,0 +1,56 @@
+on: repository_dispatch
+name: Funcbench Workflow
+jobs:
+  run_funcbench:
+    name: Running funcbench
+    if: github.event.action == 'funcbench_start'
+    runs-on: ubuntu-latest
+    env:
+      AUTH_FILE: ${{ secrets.PROMBENCH_GKE_AUTH }}
+      BRANCH: ${{ github.event.client_payload.BRANCH }}
+      BENCH_FUNC_REGEX: ${{ github.event.client_payload.BENCH_FUNC_REGEX }}
+      CLUSTER_NAME: prombench
+      GITHUB_TOKEN: ${{ secrets.PERSONAL_TOKEN }}
+      GITHUB_ORG: prometheus
+      GITHUB_REPO: prometheus
+      LAST_COMMIT_SHA: ${{ github.event.client_payload.LAST_COMMIT_SHA }}
+      PROJECT_ID: macro-mile-203600
+      PR_NUMBER: ${{ github.event.client_payload.PR_NUMBER }}
+      ZONE: europe-west3-a
+    steps:
+    - name: Update status to pending
+      run: >-
+        curl -i -X POST
+        -H "Authorization: Bearer $GITHUB_TOKEN"
+        -H "Content-Type: application/json"
+        --data '{"state":"pending","context":"funcbench-status","target_url":"https://github.com/'$GITHUB_REPOSITORY'/actions"}'
+        "https://api.github.com/repos/$GITHUB_REPOSITORY/statuses/$LAST_COMMIT_SHA"
+    - name: Prepare nodepool
+      uses: docker://prominfra/prombench:master
```

Comment:
I think we should keep prombench and funcbench completely separate and [added a follow up issue](https://github.com/prometheus/test-infra/issues/356) for this. This refactoring  will take some time so let's not block this PR and will refactor as a follow up.